### PR TITLE
Branch unit test

### PR DESCRIPTION
**Pull request by ZhangAnli**
Schedule Command Test, Edit Dog Command Test added.

### CODE REVIEW

File: src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java [hunk 1]
```diff
@@ -50,7 +54,7 @@ public void execute_allFieldsSpecified_success() {
         EditDogCommand.EditDogDescriptor descriptor = new EditDogDescriptorBuilder(editedDog).build();
```
Comment (kouyk) [inline src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java:54]: actually this one can statically import

Comment (kouyk) [reply src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java:54]: Also while you're at it, perhaps change the 2 lines above to directly retrieve ID_TWO which is the same dog, then can get rid of all the poorly named variables.

File: src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java [hunk 2]
```diff
@@ -136,7 +140,23 @@ public void execute_noFieldSpecified_success() {
     }
 
     @Test
-    public void execute_invalidDogId_failure() {
+    public void execute_getSuccessMessage_success() {
```
Comment (kouyk) [inline src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java:143]: actually isn't getSuccessMessage a protected method? How come can test ah?

File: src/test/java/dog/pawbook/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -148,12 +148,13 @@
...
     public static final EditProgramDescriptor DESC_OBEDIENCE_TRAINING;
     public static final EditProgramDescriptor DESC_POTTY_TRAINING;
+
     public static final String INVALID_NEGATIVE_ID_STRING = "-1";
     public static final String INVALID_UNKNOWN_ID_STRING = "dummy";
-
     public static final String EMPTY_STRING = "";
     public static final String WHITESPACE_STRING = "   ";
     public static final String INVALID_OUT_OF_BOUNDS_ID_STRING = String.valueOf(Long.MAX_VALUE);
+    public static final String VALID_NAME_APPLE = "Apple";
```
Comment (kouyk) [inline src/test/java/dog/pawbook/logic/commands/CommandTestUtil.java:157]: This one isn't used right?

File: src/test/java/dog/pawbook/logic/commands/ScheduleCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public void setUp() {
+        model.updateFilteredEntityList(x -> x.getValue() instanceof Program);
+    }
+
+    @Test
+    public void execute_emptyArguments_success() {
+
+        // Credit: https://www.baeldung.com/java-override-system-time
+        Clock clock = Clock.fixed(Instant.parse("2022-01-01T00:00:00Z"), ZoneId.of("UTC"));
+        LocalDate testDate = LocalDate.now(clock);
```
Comment (kouyk) [inline src/test/java/dog/pawbook/logic/commands/ScheduleCommandTest.java:44]: I think the issue with this is that our internal functional code does not make use of the same clock, it is either that we modify our code to support the use of a specified clock or we just add a program with a session date and time that happens on the same day instead, then we can use it to verify. I don't think these 2 variables are being used.

File: src/test/java/dog/pawbook/logic/commands/CommandTestUtil.java (hunk 3 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/logic/commands/CommandTestUtil.java:157]: oh yes you're right cos I changed the methods. w edit now!

Comment (ZhangAnli) [reply src/test/java/dog/pawbook/logic/commands/CommandTestUtil.java:157]: Fixed

File: src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java (hunk 2 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java:143]: Removed.

File: src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java (hunk 1 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/logic/commands/EditDogCommandTest.java]: Done.

File: src/test/java/dog/pawbook/logic/commands/ScheduleCommandTest.java (hunk 4 above)
Comment (ZhangAnli) [reply src/test/java/dog/pawbook/logic/commands/ScheduleCommandTest.java]: Fixed.

**Review by kouyk — APPROVED** [review]

**Pull request merged**